code/modules/events/creep_awakening.dm

```diff
@@ -13,7 +13,7 @@
 			continue
 		if(H.stat == DEAD)
 			continue
-		if(!H.mind.assigned_role || H.mind.assigned_role in GLOB.exp_specialmap[EXP_TYPE_SPECIAL] || H.mind.assigned_role in GLOB.exp_specialmap[EXP_TYPE_ANTAG]) //prevents ashwalkers falling in love with crewmembers they never met
+		if(!H.mind.assigned_role || !H.mind.assigned_role in GLOB.station_positions || H.mind.assigned_role in GLOB.nonhuman_positions) //only station jobs sans nonhuman roles, prevents ashwalkers falling in love with crewmembers they never met
```

Comment:
this first check might be redundant, can let me know if it's even needed or if i can cut it